src/seedu/addressbook/commands/Command.java

```diff
@@ -42,6 +42,11 @@ public static String getMessageForPersonListShownSummary(List<? extends ReadOnly
     public abstract CommandResult execute();
     //throw new UnsupportedOperationException("This method should be implement in child classes");
 
+    /**
+     *
+     * @return whether the databse is mutated
+     */
+    public abstract boolean isMutating();
```

Comment:
Nice strategy. However, one should add header commands if the behaviour of child classes are different from each other. By the way, we might not need to add a new attribute.